src/server/endpoints/searchHandler.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const HTTP_STATUSES = require('../modules/httpStatuses');
+const postsCollection = require('../collections/postsDatabase');
+const usersCollection = require('../collections/usersDatabase');
+
+function searchUsers(searchText, res) {
+  usersCollection.retrieveUsersByUsername(searchText, (result) => {
+    if (result !== null) {
+      let users = result.map(function(user) {
+        let basicUserInfo = {};
+
+        basicUserInfo.username = user.username;
+        basicUserInfo.userPicURL = user.userPicURL;
+        return basicUserInfo;
+      });
+      let obj = {people: users};
+
+      res.status(HTTP_STATUSES.OK).json(obj);
+    }
+  });
+}
+
+function searchPosts(searchText, res) {
+  postsCollection.retrievePostsByPostText(searchText, (result) => {
+    if (result !== null) {
```

Comment:
if something went wrong in the database the result will be null, so this endpoint wont answer... Errors must be handled